src/DotPulsar/Internal/PartitionedConsumer.cs

```diff
@@ -424,5 +338,224 @@ private Consumer GetConsumer(int partitionId)
                 throw new ArgumentException($"Illegal partition index in received message: ${partitionId}");
             }
         }
+
```

Comment:
both old way and new way are supported so we can easily switch between them if there is trouble

Reply:
They seem like near copy/pastes to me if the introduced `for` loop limits at one execution?  Perhaps not worth the code duplication risks.

Reply:
I want to be able to fallback to old one which works.
I will remove old one after I prove new one is okay

Reply:
I guess my thinking is we're not going to dynamically switch back to the old way without patching the affected services, so at that point we could just revert to an older version of the library if we encountered any issues.  Keeping them separate also doesn't test the 1 at a time flow in the new route until we remove the old one.  I won't force the issue, just wanted to explain where I was coming from and see if you still think this is useful.

Reply:
I was actually thinking making it configurable (this number should be tweakable anyway)